src/sage/graphs/base/static_dense_graph.pyx

```diff
@@ -378,7 +379,371 @@ def triangles_count(G):
+    if not G.is_connected():
+        raise ValueError("the input (di)graph is not connected")
+
+    for d in G.degree():
+        if d >= 8 * sizeof(unsigned long) - 1:
+            raise ValueError("the degree of the graph is too large for this method")
+
+    cdef Py_ssize_t n = G.order()
+    cdef Py_ssize_t m = G.size()
+    if min_edges is None or min_edges < n - 1:
+        min_edges = n - 1
+    if max_edges is None or max_edges > m:
+        max_edges = m
+    if min_edges > max_edges:
+        raise ValueError("we must have {} <= min_edges <= max_edges <= {}".format(n -1, m))
+
+    if n <= 1 or (not G.is_directed() and n == 2) or min_edges == m:
+        if edges_only:
+            yield list(G.edges(sort=False, labels=labels))
+        else:
+            yield G.copy()
+        return
+
+    # We map vertices to integers. We sort them by degree as a heuristic to
+    # reduce the number of operations.
+    cdef list int_to_vertex = sorted(G, key=G.degree)
+    cdef binary_matrix_t DG
+    sig_on()
+    dense_graph_init(DG, G, translation=int_to_vertex, force_undirected=True)
+
+    cdef MemoryAllocator mem = MemoryAllocator()
+    cdef int * order = <int *> mem.calloc(n, sizeof(int))
+    cdef mp_bitcnt_t * n_cpt = <mp_bitcnt_t *> mem.calloc(n, sizeof(mp_bitcnt_t))
+
+    # We use a several bitsets to store the current boundary and active neighbors.
+    # We also need another bitset that we create at the same time
+    cdef binary_matrix_t boundaries
+    binary_matrix_init(boundaries, n + 1, n)
+    cdef binary_matrix_t neighborhoods
+    binary_matrix_init(neighborhoods, n, n)
+    sig_off()
+
+    cdef bitset_t active = boundaries.rows[n]  # remaining vertices to consider
+    cdef bitset_t boundary  # neighbors of the current subset
+
+    # Initialize the process
+    cdef Py_ssize_t i = 0
+    order[0] = 0
+    bitset_complement(active, active)
+    bitset_discard(active, 0)
+    bitset_copy(neighborhoods.rows[0], DG.rows[0])
+    n_cpt[0] = 1 << bitset_len(DG.rows[0])
+
+    cdef long u, v, j
+    cdef mp_bitcnt_t c
+    cdef Py_ssize_t num_edges = 0
+    cdef list E = []
+    cdef list edges
+
+    while i >= 0:
+        sig_check()
+        if i == n - 1 and num_edges >= min_edges:
+            # yield the current solution
+            edges = [(int_to_vertex[u], int_to_vertex[v]) for le in E for u, v in le]
+            if G.is_directed():
+                yield from _yield_results_for_digraph(G, edges, edges_only, labels, min_edges, max_edges)
+            else:
+                yield _format_result(G, edges, edges_only, labels)
+
+        if n_cpt[i] > 1:
+            # Consider the next neighborhood of the current vertex.
+            # If vertex u has k active neighbors, we have 2^k possibilities. We
+            # use the binary representation of n_cpt[i] to indicate which of the
+            # k neighbors are selected. We omit the empty neighborhood which is
+            # considered else where.
```

Comment:
"else where" -> "elsewhere"

Reply:
Fixed

Reply:
Well, this iterator was already using dense data structure, so I added the functionality in it. We can of course try to do the same with another data structure and compare performances, but this should be done in a separate issue/PR.

Reply:
I see.